### DISCUSSION TITLE
How to add more memory to container on Rasppberry Pi 4?

### DISCUSSION THREAD

**Discussion by xBourner** [itzg/docker-minecraft-server#1757] [Q&A] [answered]
Hi,

i run the docker with a simple docker compose (basically all default settings, just changed name and motd) and this is the only way the server starts and not crashes for me.

1G is the only way my Pi4 starts the docker container. i have one with 4GB RAM and right now i use around 2.5G. server is okay with one person but a bit laggy. any way to make it better? 
i thought more ram would be a start but everytime i try to set a higher MEMORY value in docker-compose the server will not start or crash in the first minutes.

thx for help!

**Comment by itzg** [comment, marked as answer]
If you're not already trying setting `TYPE` to "PAPER" since that seems to be more optimized than the vanilla server.

**Reply by xBourner** [reply]
No I am on vanilla right now. Will try to use paper type. But there is nothing to tweak on ram settings?

**Reply by itzg** [reply]
You already said you tried increasing the memory. You don't have much to work with since you said 2.5 is being used. I also don't know if you're running a 64-bit OS or not.

**Reply by xBourner** [reply]
I use raspberry pi os in 64 bit. Trying to higher the value for memory environment but every value higher than 1GB will crash the server.

**Reply by itzg** [reply]
Add `DEBUG_MEMORY` set to "true" and it'll report memory available right before launching the Minecraft server.

**Reply by xBourner** [reply]
![image](https://user-images.githubusercontent.com/64064679/193445239-d13bdd37-c687-44fc-ae7e-5cf1031be011.png)

there should be something available. above is before starting and this is after starting:

![image](https://user-images.githubusercontent.com/64064679/193445288-0106aef9-17da-4a53-bd2a-fe7a202feccc.png)


EDIT: im just on 32-bit system and not-64 bit maybe this is the problem?
EDIT2: Paper version is running a bit smoother for me. but i would like to give the server a total of 1.5GB RAM

EDIT3: Paper version starts and runs for me with 1500M memory setting.
